airflow/models/taskinstance.py

```diff
@@ -375,6 +379,795 @@ def _creator_note(val):
         return TaskInstanceNote(*val)
 
 
+def _execute_task(task_instance, context, task_orig):
+    """
+    Execute Task (optionally with a Timeout) and push Xcom results.
+
+    :param task_instance: the task instance
+    :param context: Jinja2 context
+    :param task_orig: origin task
+
+    :meta private:
+    """
+    task_to_execute = task_instance.task
+
+    if isinstance(task_to_execute, MappedOperator):
+        raise AirflowException("MappedOperator cannot be executed.")
+
+    # If the task has been deferred and is being executed due to a trigger,
+    # then we need to pick the right method to come back to, otherwise
+    # we go for the default execute
+    execute_callable_kwargs = {}
+    if task_instance.next_method:
+        if task_instance.next_method:
```

Comment:
hello, coming long after the fact, but I think that during the move, this check was duplicated